c++/src/kj/async-win32.c++

```diff
@@ -133,13 +133,44 @@ private:
   Promise<IoResult> promise;
 };
 
+typedef LONG(NTAPI* NtSetInfoFuncPtr)(HANDLE, ULONG_PTR*, void*, ULONG, ULONG);
+
+NtSetInfoFuncPtr getSetInfoFileFunc()
+{
+    static thread_local NtSetInfoFuncPtr setInfoFileFunc = []() -> NtSetInfoFuncPtr {
+        HMODULE ntDll = ::GetModuleHandleA("NTDLL.DLL");
+        if (!ntDll)
+        {
+            return nullptr;
```

Comment:
You could throw an exception here, rather than `KJ_REQUIRE` that it's non-null later. That'd give more information about what went wrong.

If an exception is throw during static local initialization, the next call to the function will retry the initialization.

Reply:
Can you suggest which way of throwing an exception would be more appropriate in this case?
I'm thinking
```c++
KJ_ASSERT(ntDll != nullptr, "Could not load 'ntdll' dynamic library");
```

Reply:
`KJ_FAIL_WIN32` will also use `GetLastError()` to find out the error code and incorporate that into the exception.